### PR TITLE
Warn about unwanted mass update.

### PR DESCRIPTION
**Pull request by agoerler**
Clarify that an update without keys in data and without filter will perform a mass update.

### CODE REVIEW

File: java/working-with-cql/query-api.md [hunk 1]
```diff
@@ -972,7 +972,7 @@ Upsert doesn't allow changing the key of a child of a composition `of one`.
 
 Use the [Update](../../cds/cqn#update) statement to update existing entities with new data. The update data can be partial (patch semantics), elements without update values keep their old value, except for elements annotated with `@cds.on.update`, which are updated with the annotation value.
 
-Depending on the filter condition, the `Update` can target [individual](#update-individual-entities) or [multiple](#searched-update) entity records.
+Depending on the filter condition, the `Update` can target [](#update--entities) or [multiple](#searched-update) entity records.
```
Comment (MattSchur) [inline java/working-with-cql/query-api.md:975]:
```suggestion
Depending on the filter condition, the `Update` can target [individual](#update-individual-entities) or [multiple](#searched-update) entity records.
```

File: java/working-with-cql/query-api.md [hunk 2]
```diff
@@ -1010,6 +1010,10 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
    .data("title", "CAP Matters");
 ```
 
+::: warning
```
Comment (MattSchur) [inline java/working-with-cql/query-api.md:1013]:
```suggestion
::: danger
```

File: java/working-with-cql/query-api.md [hunk 3]
```diff
@@ -1010,6 +1010,10 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
...
+If a key value is neither contained in the data nor specified by a filter (`where`, `byId`, `matching`) a [searched update]([url](#searched-update)) will be performed, which updates _all_ entities with the given data.
```
Comment (MattSchur) [inline java/working-with-cql/query-api.md:1014]:
```suggestion
If key values are neither contained in the data nor specified by a filter (`where`, `byId`, `matching`) a [searched update](#searched-update) is performed, which updates _all_ entities matching the filter condition with the given data.
```

File: java/working-with-cql/query-api.md [hunk 4]
```diff
@@ -1010,6 +1010,10 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
    .data("title", "CAP Matters");
 ```
 
+::: danger
+If a key value is neither contained in the data nor specified by a filter (`where`, `byId`, `matching`) a [searched update]([url](#searched-update)) will be performed, which updates _all_ entities with the given data.
```
Comment (agoerler) [inline java/working-with-cql/query-api.md:1014]:
```suggestion
If a key value are not contained in the data and no filter (`where`, `byId`, `matching`) is specified a [searched update]([url](#searched-update)) will be performed, which updates _all_ entities with the given data.
```

File: java/working-with-cql/query-api.md (hunk 3 above)
Comment (agoerler) [reply java/working-with-cql/query-api.md:1014]: or

```suggestion
If key values are not contained in the data and no filter (`where`, `byId`, `matching`) is specified a [searched update]([url](#searched-update)) will be performed, which updates _all_ entities with the given data.
```

File: java/working-with-cql/query-api.md (hunk 4 above)
Comment (MattSchur) [reply java/working-with-cql/query-api.md:1014]: or just
```suggestion
If the key values are not contained in the data and no filter (`where`, `byId`, `matching`) is specified an unconditional update is performed, which updates _all_ DB entries of the entity with the given data.
```

File: java/working-with-cql/query-api.md [hunk 5]
```diff
@@ -1010,6 +1010,10 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
    .data("title", "CAP Matters");
 ```
 
+::: danger
+If key values are not contained in the data and no filter (`where`, `byId`, `matching`) is specified a [searched update]([url](#searched-update)) will be performed, which updates _all_ entities with the given data.
```
Comment (MattSchur) [inline java/working-with-cql/query-api.md:1014]:
```suggestion
If key values are not contained in the data and no filter (`where`, `byId`, `matching`) is specified a [searched update](#searched-update) is performed, which updates _all_ entities with the given data.
```

**Review by MattSchur — APPROVED** [review]

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**